src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -175,20 +191,34 @@ void HexagonGame::draw()
             playerTris3D.unsafe_emplace_other(playerTris);
         }
 
+        if(renderAbove)
+        {
+            for(std::size_t i = 0; i < aboveMain; ++i)
+            {
+                wallQuads3DTop.unsafe_emplace_other(wallQuads);
+                pivotQuads3DTop.unsafe_emplace_other(pivotQuads);
+                playerTris3DTop.unsafe_emplace_other(playerTris);
+            }
+        }
+
         const auto adjustAlpha = [&](sf::Color& c, const float i)
         {
             SSVOH_ASSERT(styleData._3dAlphaMult != 0.f);
 
             const float newAlpha =
                 (static_cast<float>(c.a) / styleData._3dAlphaMult) -
-                i * styleData._3dAlphaFalloff;
+                (styleData._3dAlphaMirror ? std::abs(i + 1) : i) *
+                    styleData._3dAlphaFalloff;
 
             c.a = Utils::componentClamp(newAlpha);
         };
 
-        for(int j(0); j < static_cast<int>(depth); ++j)
+        for(int j(0);
+            j < static_cast<int>(renderAbove ? depth + aboveMain : depth); ++j)
         {
-            const float i(depth - j - 1);
+            const bool renderingAbove(j >= depth && renderAbove);
+            const float jAdj(j - depth * renderingAbove * (depth >= 0));
+            const float i(depth - j - 1 + layerOffset);
```

Comment:
Ditto, especially confused about `jAdj`.

Reply:
Yeah, it's a bit confusing because i didn't want to split it into two loops, it's so it can switch to the next vertex container in the same loop. It's probably a better idea to just split it into two loops.